src/main.rs

```diff
@@ -18,6 +20,35 @@ use rinja::Template;
 
 use crate::guides::{get_prev_next_link, parse_guides};
 
+pub(crate) const LATEST_VERSION: &'static str = "v0.1";
+
+pub(crate) fn get_all_versions() -> Vec<String> {
+    let manifest_dir = std::env::var("CARGO_MANIFEST_DIR").expect("CARGO_MANIFEST_DIR not set");
+    let path = Path::new(&manifest_dir).join("src").join("md-pages");
+    let dir = std::fs::read_dir(path).expect("Failed to read md-pages directory");
+    let versions: Vec<_> = dir
+        .filter_map(|entry| {
+            if let Ok(entry) = entry {
+                if entry.path().is_dir() {
+                    return Some(entry.file_name().to_string_lossy().to_string());
+                }
+            };
+
+            None
+        })
+        .collect();
+
+    [String::from("latest")]
+        .into_iter()
+        .chain(
+            versions
+                .into_iter()
+                .rev()
+                .skip_while(|dir| dir != LATEST_VERSION),
+        )
+        .collect()
+}
```

Comment:
I'd love to have this memoized instead of happening on each request, but it's a minor thing.

Reply:
This won't work. We don't ship the `md-pages` directory in production, so this will simply fail when deployed. There's also no guarantee on the order of the directories returned by `fs::read_dir`, but this is secondary.

I don't understand why we even need this. We already hardcode the list of versions in `get_categories_for_version`, no?